src/pages/CookieClicker.vue

```diff
@@ -1,49 +1,263 @@
+  { name: 'Grandma', price: 100, cps: 1, count: 0 },
+  { name: 'Farm', price: 1100, cps: 8, count: 0 },
 ]);
 
+// CPS Kalkulatsioon
+const cps = computed(() => {
+  return buildings.value.reduce((total, building) => {
+    return total + building.cps * building.count;
+  }, 0);
+});
+
+//golden cookie system
+let goldenCookies = ref([]);
+let goldenBonusActive = ref(false);
+
+// Upgrade System 
+let upgrades = ref([
+  {
+    name: "Better Clicks",
+    cost: 100,
+    effect: () => { clickPower.value += 1; }, // Ensure `clickPower.value`
+    bought: false,
+    description: "Increases cookies per click by +1."
+  },
+  {
+    name: "Faster Grandmas",
+    cost: 500,
+    effect: () => { buildings.value[1].cps *= 2; },
+    bought: false,
+    description: "Doubles Grandma's CPS."
+  },
+  {
+    name: "Cheaper Buildings",
+    cost: 1000,
+    effect: () => { buildings.value.forEach(b => b.price = Math.ceil(b.price * 0.9)); },
+    bought: false,
+    description: "Reduces all building prices by 10%."
+  }
+]);
+
+let clickPower = ref(1); // Default click power alguseks 1
+
+// Laeb local storage
+function loadGame() {
+  let save = JSON.parse(localStorage.getItem('cookieClickerSave'));
+  if (save) {
+    cookies.value = save.cookies ?? 0;
+    buildings.value = save.buildings ?? buildings.value;
+    goldenBonusActive.value = save.goldenBonusActive ?? false;
+    clickPower.value = save.clickPower ?? 1;
+    
+    // saved upgrades
+    if (save.upgrades) {
+      upgrades.value = save.upgrades.map(savedUpgrade => {
+        let original = upgrades.value.find(u => u.name === savedUpgrade.name);
+        return original ? { ...savedUpgrade, effect: original.effect } : savedUpgrade;
+      });
+    }
+  }
+}
+
+// save func
+function saveGame() {
+  localStorage.setItem('cookieClickerSave', JSON.stringify({
+    cookies: cookies.value,
+    buildings: buildings.value,
+    goldenBonusActive: goldenBonusActive.value,
+    upgrades: upgrades.value,
+    clickPower: clickPower.value
+  }));
+}
+
+// autosave iga 10 sek
+setInterval(saveGame, 10000);
+
+//earn cookies over time
 setInterval(() => {
-    cookies.value += cps.value;
+  cookies.value += cps.value;
 }, 1000);
 
-function cookieClick(){
-    cookies.value += 1;
+// Click Cookie arvestab clickpower upgradega
+function cookieClick() {
+  cookies.value += clickPower.value;
+  saveGame();
 }
 
-function buyBuilding(building){
+// buy building
+function buyBuilding(building) {
+  if (cookies.value >= building.price) {
     cookies.value -= building.price;
     building.count++;
-    building.price += Math.ceil((building.price / 100) * 15)
+    building.price += Math.ceil((building.price / 100) * 15);
+    saveGame();
+  }
 }
 
-const cps = computed(() => {
-    return buildings.value.reduce((total, building) => {
-        return total + building.cps * building.count;
-    }, 0)
-});
+// Golden Cookie surprise
+function clickGoldenCookie(id) {
```

Comment:
cool function, but you could make it better to add new stuff in future by making it like % based and have it call different effect functions instead.